static/src/javascripts/projects/commercial/modules/liveblog-adverts.js

```diff
@@ -69,34 +69,28 @@ const getSlotName = (isMobile, slotCounter) => {
 	return `inline${slotCounter + 1}`;
 };
 
-const insertAds = (slots) => {
+const insertAds = (paras) => {
 	const isMobile = getBreakpoint() === 'mobile';
 
-	for (let i = 0; i < slots.length && SLOTCOUNTER < MAX_ADS; i += 1) {
-		const slotName = getSlotName(isMobile, SLOTCOUNTER);
-
-		const adSlots = createSlots('inline', {
-			name: slotName,
-			classes: 'liveblog-inline',
-		});
-
-		adSlots.forEach((adSlot) => {
-			if (slots[i] && slots[i].parentNode) {
-				slots[i].parentNode.insertBefore(adSlot, slots[i].nextSibling);
-			}
-		});
-
-		// Only add the first adSlot (the DFP one) in DFP/GTP
-		if (slots[i] && slots[i].parentNode) {
-			addSlot(adSlots[0], false);
-			SLOTCOUNTER += 1;
+	for (let i = 0; i < paras.length && AD_COUNTER < MAX_ADS; i += 1) {
+		const para = paras[i];
+		if (para && para.parentNode) {
+			const adSlot = createAdSlot('inline', {
+				name: getSlotName(isMobile, AD_COUNTER),
+				classes: 'liveblog-inline',
+			});
+			// insert the ad slot container into the DOM
+			para.parentNode.insertBefore(adSlot, para.nextSibling);
+			// load and display the advert via GAM
+			addSlot(adSlot, false);
+			AD_COUNTER += 1;
```

Comment:
Tidied this up @chrislomaxjones

Reply:
Great,`slot` -> `para` makes this so much easier to understand.

Reply:
Types would be great, here, wouldn’t they?

Can the paras be things like `<blockquote>`, `<embed>`, `<picture>` or other?

Reply:
Types would be very helpful - yes!

`paras` can be any element in theory - it depends on the `rules` selector properties passed into `spaceFiller` e.g.

https://github.com/guardian/frontend/blob/498120e31f3a4463183df876b9fd2b4c70325f2d/static/src/javascripts/projects/commercial/modules/liveblog-adverts.js#L49-L60

`paras` is just the naming convention used in other `spaceFiller` clients so I followed that here.

Reply:
So `paras` are the candidates that win a round of Spacefinder, so `liveblocks` may be more accurate? `liveblocks` matches the DCR name for these elements, I'm not sure what they're called in frontend.

Reply:
`elements` ?

Reply:
`elements` makes sense to me as they could be element - but I assume in reality they are paragraph boundaries in most cases.

I think however this is venturing into another PR for a wider tidy up of space-finder nomenclature...?